src/components/logger/logger.js

```diff
@@ -49,30 +49,21 @@ export default class Logger extends React.Component {
   static propTypes = {
     width: PropTypes.number.isRequired,
     height: PropTypes.number.isRequired,
-    logs: PropTypes.array.isRequired,
+    logs$: PropTypes.object,
   };
 
   state = {
     events: [],
   };
 
   onClear = () => this.setState({ events: [] });
+  onEvent = (event) => this.setState((state) => ({ events: [...state.events, event] }));
 
-  componentWillUpdate({ logs }) {
-    this.logsSubscriptions.map((sub) => sub.unsubscribe());
-    this.logsSubscriptions = logs.map((obs) => {
-      return obs.subscribe((event) => {
-        this.setState((state) => ({ events: [...state.events, event] }));
-      });
-    });
-  }
-
-  componentDidMount() {
-    this.logsSubscriptions = this.props.logs.map((obs) => {
-      return obs.subscribe((event) => {
-        this.setState((state) => ({ events: [...state.events, event] }));
-      });
-    });
+  componentDidUpdate(prevProps) {
```

Comment:
you should leave componentDidMount for the case when you initially receive a stream and have to subscribe to it (I know that it's not the case right now, but can be in the future)